src/api/api.ts

```diff
@@ -0,0 +1,24 @@
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { client } from '../utils/fetchClient';
+
+export const getPostsData = (userId: number) => {
+  return client.get<Post[]>(`/posts?userId=${userId}`);
+};
+
+export const getUsersData = () => {
+  return client.get<User[]>('/users');
+};
+
+export const getCommentsData = (postId: number) => {
+  return client.get<Comment[]>(`/comments?postId=${postId}`);
+};
+
+export const postCommentData = (data: Omit<Comment, 'id'>) => {
+  return client.post<Comment>('/comments', data);
+};
+
+export const deleteCommentData = (id: number) => {
+  return client.delete(`/comments/${id}`);
+};
```

Comment:
again. correct naming is very important thing
```suggestion
export const getPostsData = (userId: number) => {
  return client.get<Post[]>(`/posts?userId=${userId}`);
};

export const getUsers = () => {
  return client.get<User[]>('/users');
};

export const getComments = (postId: number) => {
  return client.get<Comment[]>(`/comments?postId=${postId}`);
};

export const createComment = (newComment: Omit<Comment, 'id'>) => {
  return client.post<Comment>('/comments', newComment);
};

export const deleteComment = (id: number) => {
  return client.delete(`/comments/${id}`);
};
```